gpflow/base.py

```diff
@@ -192,55 +237,55 @@ def assign(
         )
 
     @property
-    def is_tensor_like(self):
+    def is_tensor_like(self) -> bool:
         """
         This method means that TensorFlow's `tensor_util.is_tensor` function
         will return `True`
         """
         return True
 
     @property
-    def name(self):
+    def name(self) -> str:
         return self._unconstrained.name
 
     @property
-    def initializer(self):
+    def initializer(self) -> UNKNOWN:
```

Comment:
What's the value of this over leaving it as
```suggestion
    def initializer(self):
```
?